analyzers/nodejs/nodejs.go

```diff
@@ -215,7 +215,7 @@ func (a *Analyzer) Analyze() (graph.Deps, error) {
 
 	// if npm as a tool does not exist, skip this
 	if a.NPM.Exists() {
-		pkgs, err := a.NPM.List(filepath.Dir(a.Module.BuildTarget))
+		pkgs, err := a.NPM.List(a.Module.BuildTarget)
```

Comment:
To ease some of your concerns this won't break catostrophically on old config files with a trailing slash, however it will break with a `package.json`.

I agree with this and think both of these cases are the only times we should issue a warning or change the input. If you have any other cases let me know.

Reply:
Updated the logic to sanitize `a.Module.BuildTarget` in `Analyze()` as well as `IsBuilt()`. Looking for feedback on the warning messages in particular. I think they are a little verbose but I am highly favoring descriptive information here to make the changes easier for users.